lib/manifest.js

```diff
@@ -0,0 +1,78 @@
+let path = require("path");
+let { createFile, generateFingerprint, uriJoin } = require("faucet-pipeline-util");
+let retry = require("./retry");
+let retryTimings = [10, 50, 100, 250, 500, 1000];
+
+class Manifest {
```

Comment:
That name doesn't quite work for my mental model of how this all fits together: In my head (and unfinished code), the manifest is just one potential (i.e. optional!) artifact resulting from file operations.

I'll be more concrete once I get around to, you know, actually writing code again.

Reply:
Agreed. I don't really like `AssetManager` though...